### ISSUE TITLE
Support for additional authorization types

### ISSUE THREAD

**Issue by jeremyowensboggs** [jakobht/azurex#46]
Currently, there is only shared key support.  We also need support for SAS tokens and Managed Identities.  What do you think that should look like?  
1) Do we want to handle it all in config, so the call signatures stay the same?
2) add support for auth type in options?
3) a `use Azurex, <options>` approach?
4) other?

Thank You

**Comment by jakobht** [comment]
I don't think we should change the signatures. I think a combination of 2 and 3 would be great. Not sure how big of a change it is to support other auth?

**Comment by nomtrosk** [comment]
I gave it a swing by introducing authentication using Service Principals here  https://github.com/jakobht/azurex/pull/56. I'm open for comments!

**Comment by TomBeckett** [comment]
@jakobht any comments? LGTM.

Being able to Managed Identities is a huge security improvement!

**Comment by ktayah** [comment]
Took @nomtrosk work with Service Principals and started extending it to work with Managed Identities as well. Not sure how we want to manage merging it all together. I can add a PR to merge my stuff into yours @nomtrosk if you like.

**Comment by nomtrosk** [comment]
@ktayah Yes, please make a PR to my fork
